test/WriterTest.cpp

```diff
@@ -18,15 +18,15 @@ BOOST_AUTO_TEST_SUITE(WriterSuite)
 BOOST_AUTO_TEST_CASE(Writer0)
 {
     Reader r("test.mae");
-    auto w = new Writer("test_write.mae");
+    auto w = std::make_shared<Writer>("test_write.mae");
     std::vector<std::shared_ptr<Block> > input;
 
     std::shared_ptr<Block> b;
     while ((b = r.next(CT_BLOCK)) != nullptr) {
         input.push_back(b);
         w->write(b);
     }
-    delete w;
+    w.reset();
```

Comment:
`w.reset()` seems superfluous.

Reply:
I want to release the shared_ptr so that it will delete the underlying object doing the writing, including the file handles it owns.  This guarantees buffered writing will be flushed.